e2e-tests/src/test/button_game/mod.rs

```diff
@@ -55,6 +60,56 @@ pub fn marketplace(config: &Config) -> Result<()> {
     )?;
     marketplace.buy(&sign(&conn, player), None)?;
 
+    // Upgrade begins
+
+    // Performing code upgrade
+    let set_code_result = marketplace.set_code(
+        &sign(&conn, &authority),
+        config
+            .test_case_params
+            .marketplace_v2_code_hash
+            .as_ref()
+            .expect("New code's code_hash must be specified."),
+        None, // Alternative for performing "atomic" upgrade + migration (make sure that selector to migrate method is correct)
+              // Some("0x060d3f50".to_string())
+    );
+    info!("Trying to set code actual hash_code: {:?}", set_code_result);
+    assert!(set_code_result.is_ok());
```

Comment:
Suggestion: I think inline comments like this really reduce readability. If you feel like this code needs more explanation, extract it to a function and add a docstring to that function.